src/Runner.Listener/MessageListener.cs

```diff
@@ -228,6 +240,24 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
                     // Decrypt the message body if the session is using encryption
                     message = DecryptMessage(message);
 
+                    
+                    if (message != null && message.MessageType == BrokerMigrationMessage.MessageType)
+                    {
+                        Trace.Info("BrokerMigration message received. Polling Broker for messages...");
+
+                        var migrationMessage = JsonUtility.FromString<BrokerMigrationMessage>(message.Body);                        
+                        var brokerServer = HostContext.GetService<IBrokerServer>();
+
+                        await brokerServer.ConnectAsync(migrationMessage.BrokerBaseUrl, _creds);
+                        message = await brokerServer.GetRunnerMessageAsync(token, 
```

Comment:
If the runner ever polls with the wrong sessionID for its runnerID, we'd planned for Broker to respond [403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403). Am I right that that 403 would be caught [here in BrokerHttpClient](https://github.com/actions/runner/blob/40f813e0fb2a3b500858c907723f3efabc9d979b/src/Sdk/WebApi/WebApi/BrokerHttpClient.cs#L123-L126)?

Reply:
If so, Runner should try to recreate the session after a 403. (Maybe that already happens somewhere in Runner's code, like I said I'm new to the runner.)

Reply:
If we get a 403, the runner will exit and restart and start the whole `MessageListener` flow over again, thus attempting to create a new sessions

Reply:
@luketomlinson Just a heads up that we actually have the broker responding `409 Conflict` in this case: https://github.com/github/actions-broker-listener/pull/1031/files#diff-37c615ad33e1d15f935093bd9fc81ccecce06ed47240d27cacce9ced12d79ee3R303